src/components/home/modals/AliasAddress.vue

```diff
@@ -0,0 +1,303 @@
+          <b-button
+            block
+            class="btn-add"
+            variant="warning"
+            size="lg"
+            @click="addAlias">Add
+          </b-button>
+        </b-col>
+      </b-row>
+    </div>
+  </b-modal>
+</template>
+
+<script>
+import Vue from 'vue'
+export default {
+    name: 'AliasAddress',
+    props: {
+        addresses: {
+            type: Object,
+            require: true,
+            default: function() {}
+        },
+        address: {
+            type: String,
+            require: true,
+            default: ''
+        }
+    },
+    data() {
+        return {
+            curAlias: '',
+            curAddress: '',
+            aliasNum: 3,
+            aliasAddresses: {}
+        }
+    },
+    created() {
+        this.aliasAddresses = JSON.parse(window.localStorage.getItem(this.address)).alias
+    },
+    computed: {
+        isValidAddress() {
+            if (!this.curAddress) {
+                return void 0
+            }
+            return this.isExistedAddress
+        },
+        isExistedAddress() {
+            return this.curAddress in this.addresses
+        },
+        isExistedAlias() {
+            let alias = JSON.parse(window.localStorage.getItem(this.address)).alias
+            for (let key in alias) {
+                if (this.curAlias === alias[key]) {
+                    return true
+                }
+            }
+            return false
+        },
+        isAliasLessEight() {
+            return this.curAlias.length < 9
+        },
+        isValidAlias() {
+            if (!this.curAlias) {
+                return void 0
+            }
+            return this.isAliasLessEight && !this.isExistedAlias
+        },
+        from3rdParty() {
+            return false
+        }
+    },
+    methods: {
+        resetData() {
+        },
+        deleteAlias(addr) {
+            let userInfo = JSON.parse(window.localStorage.getItem(this.address))
+            delete userInfo.alias[addr]
+            window.localStorage.setItem(this.address, JSON.stringify(userInfo))
+            this.aliasAddresses = userInfo.alias
+        },
+        closeModal() {
+            this.$refs.aliasAddressModal.hide()
+        },
+        copyText(buttonId, addrToCopy, index) {
+            this.$refs[addrToCopy][index].select()
+            window.document.execCommand('copy')
+            this.$root.$emit('bv::show::popover', buttonId)
+            setTimeout(() => {
+                this.$root.$emit('bv::hide::popover', buttonId)
+            }, 400)
+        },
+        addAlias() {
+            let userInfo = JSON.parse(window.localStorage.getItem(this.address))
+            Vue.set(userInfo.alias, this.curAddress, this.curAlias)
+            window.localStorage.setItem(this.address, JSON.stringify(userInfo))
+            this.aliasAddresses = JSON.parse(window.localStorage.getItem(this.address)).alias
+            this.$refs.aliasAddressModal.hide()
```

Comment:
Do not close Modal until user click "Close". User may want to add multiple alias address.